common/containers/Tabs/RestoreKeystore/components/KeystoreDetails.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { Component } from 'react';
+import Template from './Template';
+import { fromPrivateKey, IFullWallet } from 'ethereumjs-wallet';
+import { makeBlob } from 'utils/blob';
+import './KeystoreDetails.scss';
+
+interface State {
+  secretKey: string;
+  password: string;
+  fileName: string;
+  isPasswordVisible: boolean;
+  isPrivateKeyVisible: boolean;
+  wallet: IFullWallet | null | undefined;
+}
+const initialState: State = {
+  secretKey: '',
+  password: '',
+  isPasswordVisible: false,
+  isPrivateKeyVisible: false,
+  fileName: '',
+  wallet: null
+};
+
+class KeystoreDetails extends Component<{}, State> {
+  public state = initialState;
+
+  public componentWillUnmount() {
+    this.resetState();
+  }
+  public render() {
+    const {
+      secretKey,
+      isPasswordVisible,
+      isPrivateKeyVisible,
+      password,
+      wallet,
+      fileName
+    } = this.state;
+    const passwordValid = password.length > 9;
+    const privateKeyValid =
```

Comment:
ethereumjs-util has a validator for this

Reply:
isValidPrivate in ethereumjs-util expects a buffer - which means we would have to convert the hex string to a buffer to do the check, which in the case that the hex string is invalid, that would throw an Error on the conversion.